### PR TITLE
Change the numerical test to be more comprehensive.

### LINKED ISSUE #19267: Fix numerical tests for all backends
Re-enable the following line in `numerical_test.py`.
https://github.com/keras-team/keras/pull/19260/files#diff-21b406be9f4fa0e56ac7bffb71793ddab1c9811503b7763792dcb160426ebb39L44-R48

### PR DESCRIPTION
**Pull request by haifeng-jin**
Resolves #19267.

All numerical issues are fixed.
Add `BatchNormalization` to `numerical_tests.py`.
Also test weights, predict, evaluate, in addition to training history.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
Nice! LGTM
